test-suite-http-client-javanet-ssl/src/test/groovy/io/micronaut/http/client/javanet/SslStaticCertSpec.groovy

```diff
@@ -0,0 +1,80 @@
+package io.micronaut.http.client.javanet
+
+import io.micronaut.context.ApplicationContext
+import io.micronaut.context.annotation.Requires
+import io.micronaut.context.env.Environment
+import io.micronaut.core.io.socket.SocketUtils
+import io.micronaut.http.HttpRequest
+import io.micronaut.http.HttpResponse
+import io.micronaut.http.annotation.Controller
+import io.micronaut.http.annotation.Get
+import io.micronaut.http.client.HttpClient
+import io.micronaut.runtime.server.EmbeddedServer
+import reactor.core.publisher.Flux
+import spock.lang.Shared
+import spock.lang.Specification
+
+class SslStaticCertSpec extends Specification {
+
+    @Shared
+    String host = Optional.ofNullable(System.getenv(Environment.HOSTNAME)).orElse(SocketUtils.LOCALHOST)
+
+    ApplicationContext context
+    EmbeddedServer embeddedServer
+    HttpClient client
+
+    void setup() {
+        context = ApplicationContext.run([
+                'spec.name': 'SslStaticCertSpec',
+                'micronaut.ssl.enabled': true,
+                'micronaut.ssl.keyStore.path': 'classpath:keystore.p12',
+                'micronaut.ssl.keyStore.password': 'foobar',
+                'micronaut.ssl.keyStore.type': 'PKCS12',
+                'micronaut.ssl.protocols': ['TLSv1.2'],
+                'micronaut.server.ssl.port': -1,
+                'micronaut.ssl.ciphers': ['TLS_RSA_WITH_AES_128_CBC_SHA',
+                                          'TLS_RSA_WITH_AES_256_CBC_SHA',
+                                          'TLS_RSA_WITH_AES_128_GCM_SHA256',
+                                          'TLS_RSA_WITH_AES_256_GCM_SHA384',
+                                          'TLS_DHE_RSA_WITH_AES_128_GCM_SHA256',
+                                          'TLS_DHE_RSA_WITH_AES_256_GCM_SHA384',
+                                          'TLS_DHE_DSS_WITH_AES_128_GCM_SHA256',
+                                          'TLS_DHE_DSS_WITH_AES_256_GCM_SHA384'],
+                'micronaut.http.client.ssl.insecure-trust-all-certificates': true
+        ])
+        embeddedServer = context.getBean(EmbeddedServer).start()
+        client = context.createBean(HttpClient, embeddedServer.getURL())
+    }
+
+    void cleanup() {
+        client.close()
+        context.close()
+    }
+
+    void "expect the url to be https"() {
+        expect:
+        embeddedServer.getURL().toString() == "https://${host}:${embeddedServer.port}"
+    }
+
+    void "test send https request"() {
+        when:
+        Flux<HttpResponse<String>> reactiveSequence = Flux.from(client.exchange(
```

Comment:
why not just write the test with `client.toBlocking` instead of the `Flux`. Are we testing anything that we need to use a reactive sequence in the test?

Reply:
We have 2 clients effectively, calling `toBlocking` gives you a `JavanetBlockingHttpClient` which has a different `exchange` method to the non-blocking version.

The client TCK should be updated to check both, currently it only tests the blocking one IIRC

Reply:
https://github.com/micronaut-projects/micronaut-core/pull/8790